extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/HibernateOrmRecorder.java

```diff
@@ -92,8 +93,18 @@ public DataSourceTenantConnectionResolver get() {
         };
     }
 
-    public Supplier<JPAConfig> jpaConfigSupplier(HibernateOrmRuntimeConfig config) {
-        return () -> new JPAConfig(config);
+    public Supplier<JPAConfig> jpaConfigSupplier(HibernateOrmRuntimeConfig config, ShutdownContext shutdownContext) {
+        return () -> {
+            JPAConfig jpaConfig = new JPAConfig(config);
+            shutdownContext.addShutdownTask(
+                    ShutdownContext.DEFAULT_PRIORITY + 1, new Runnable() {
```

Comment:
See this is precisely where more descriptive constants would help understand what we're doing :)